### PR TITLE
*: FLUSH TABLE statement front end

### PR DESCRIPTION
**Pull request by tiancaiamao**
try to support mydumper, which send `FLUSH TABLES WITH READ LOCK`
https://github.com/pingcap/tidb/issues/1690

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -4110,6 +4117,48 @@ ShowTableAliasOpt:
...
+NoWriteToBinLogAliasOpt:
+	"NO_WRITE_TO_BINLOG"
+	{
+		$$ = true
+	}
+|	"LOCAL"
+	{
+		$$ = true
+	}
+|
```
Comment (shenli) [inline parser/parser.y]: Put this to the first place.

File: parser/parser.y [hunk 2]
```diff
@@ -4110,6 +4117,48 @@ ShowTableAliasOpt:
...
+	{
+		$$ = $1
+	}
+
+WithReadLockOpt:
+	"WITH" "READ" "LOCK"
+	{
+		$$ = true
+	}
+|	{
```
Comment (shenli) [inline parser/parser.y]: Put this to the first place.

**Comment by shenli** [comment]
Rest LGTM
@coocood @zimulala PTAL

File: parser/parser_test.go [hunk 3]
```diff
@@ -384,6 +301,112 @@ AAAAAAAAAAAA5gm5Mg==
...
+		// For show full columns
+		{"show columns in t;", true},
+		{"show full columns in t;", true},
+		// For show create table
+		{"show create table test.t", true},
+		{"show create table t", true},
+
+		// set
+		// user defined
+		{"SET @a = 1", true},
```
Comment (coocood) [inline parser/parser_test.go]: SET statement is not DBA statement.

File: executor/executor_simple.go [hunk 4]
```diff
@@ -68,6 +68,8 @@ func (e *SimpleExec) Next() (*Row, error) {
 	switch x := e.Statement.(type) {
 	case *ast.UseStmt:
 		err = e.executeUse(x)
+	case *ast.FlushTableStmt:
+		err = e.executeFlushTable(x)
 	case *ast.SetStmt:
```
Comment (coocood) [inline executor/executor_simple.go]: No `*ast.FlushStmt` case?

File: executor/executor_simple.go [hunk 5]
```diff
@@ -68,6 +68,8 @@ func (e *SimpleExec) Next() (*Row, error) {
 	switch x := e.Statement.(type) {
 	case *ast.UseStmt:
 		err = e.executeUse(x)
+	case *ast.FlushTableStmt:
+		err = e.executeFlushTable(x)
```
Comment (coocood) [inline executor/executor_simple.go]: No `*ast.FlushStmt` case?

File: plan/planbuilder.go [hunk 6]
```diff
@@ -95,6 +95,8 @@ func (b *planBuilder) build(node ast.Node) Plan {
 		return b.buildInsert(x)
 	case *ast.LoadDataStmt:
 		return b.buildLoadData(x)
+	case *ast.FlushTableStmt:
+		return b.buildSimple(x)
```
Comment (coocood) [inline plan/planbuilder.go]: Ditto

File: parser/parser_test.go (hunk 3 above)
Comment (tiancaiamao) [inline parser/parser_test.go]: SET *is* DBA statement, you can find it here:

https://dev.mysql.com/doc/refman/5.7/en/set-statement.html

table of contents on the left shows:

```
SQL Statement Syntax
>    Database Administration Statements
      >    SET Syntax
            >    SET Syntax for Variable Assignment
       
```

File: executor/executor_simple.go (hunk 5 above)
Comment (tiancaiamao) [inline executor/executor_simple.go]: no general `FlushStmt` in this PR yet, just add `FlushTableStmt`, which is a special case of the FLUSH syntax.

File: plan/planbuilder.go (hunk 6 above)
Comment (tiancaiamao) [inline plan/planbuilder.go]: Ditto

**Comment by coocood** [comment]
LGTM

File: plan/planbuilder.go (hunk 6 above)
Comment (shenli) [inline plan/planbuilder.go]: Keep it in alphabet order.

**Comment by shenli** [comment]
LGTM

**Pull request merged**